core/src/banking_stage/transaction_scheduler/scheduler_controller.rs

```diff
@@ -160,13 +198,133 @@ impl SchedulerController {
                 max_age_slot: last_slot_in_epoch,
             };
             let transaction_priority_details = packet.priority_details();
-            self.container.insert_new_transaction(
+            if self.container.insert_new_transaction(
                 transaction_id,
                 transaction_ttl,
                 transaction_priority_details,
-            );
+            ) {
+                self.count_metrics.num_dropped_on_capacity += 1;
+            }
+            self.count_metrics.num_buffered += 1;
+        }
+    }
+}
+
+#[derive(Default)]
+struct SchedulerCountMetrics {
+    interval: AtomicInterval,
+
+    /// Number of packets received.
+    num_received: usize,
+    /// Number of packets buffered.
+    num_buffered: usize,
+
+    /// Number of transactions scheduled.
+    num_scheduled: usize,
+    /// Number of completed transactions received from workers.
+    num_finished: usize,
+    /// Number of transactions that were retryable.
+    num_retryable: usize,
+
+    /// Number of transactions that were immediately dropped on receive.
+    num_dropped_on_receive: usize,
+    /// Number of transactions that were dropped due to sanitization failure.
+    num_dropped_on_sanitization: usize,
+    /// Number of transactions that were dropped due to clearing.
+    num_dropped_on_clear: usize,
+    /// Number of transactions that were dropped due to exceeded capacity.
+    num_dropped_on_capacity: usize,
+}
```

Comment:
Would be possible to report stats for following topics, or do not belong?
- respecting priority (maybe counter for "just scheduled tx has prio higher than lowest in-fly transactions", or something else, so we can see realtime how prio is respected.) 
- workers load balancing
- block quality (cost_track probably reports total CU and total TXs packed into a block, maybe can add total prio in a block)

Reply:
- respecting priority - this is probably a bigger change since we don't currently track which txs hold each account lock with what priority. So we'd need someway to track min outstanding priority for each account at least.
- workers load balancing - there's a separate PR (#33895) which adds detailed metrics per worker. I think that should probably be sufficient for checking any load-balancing.
- block quality - this is a good idea. Want to try to think how I could also back that into the existing impl, and then we can monitor both as we roll out since the scheduler is intended to help increase that, among other things.

Reply:
Going to add a metric in the QoS service which tracks total fees and priority fees collected.
Will be separate from this PR though, because it makes several unrelated changes and also affects the thread-local-multi-iterator banking stage.